### ISSUE TITLE
h5read() regression between rhdf5 2.26 and 2.7

### ISSUE THREAD

**Issue by hpages** [Huber-group-EMBL/rhdf5#35]
In recent rhdf5 2.27, `h5read()` no longer seems to handle empty indices:
```
library(rhdf5)
h5createFile("test.h5")
m <- matrix(1:15, ncol=3)
h5write(m, "test.h5", "m")
h5read("test.h5", "m", list(integer(0), NULL))
# Error in indices[[j]] : subscript out of bounds
h5read("test.h5", "m", list(integer(0), integer(0)))
# Error in indices[[j]] : subscript out of bounds
```
`sessionInfo()`:
```
R Under development (unstable) (2019-02-07 r76070)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.5 LTS

Matrix products: default
BLAS: /home/hpages/R/R-3.6.r76070/lib/libRblas.so
LAPACK: /home/hpages/R/R-3.6.r76070/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] rhdf5_2.27.12

loaded via a namespace (and not attached):
[1] compiler_3.6.0 Rhdf5lib_1.5.1
```
This was working with rdhf5 2.26:
```
library(rhdf5)
h5read("test.h5", "m", list(integer(0), NULL))
#      [,1] [,2] [,3]
h5read("test.h5", "m", list(integer(0), integer(0)))
# <0 x 0 matrix>
```
`sessionInfo()`:
```
R version 3.5.1 Patched (2018-08-01 r75051)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.5 LTS

Matrix products: default
BLAS: /home/hpages/R/R-3.5.r75051/lib/libRblas.so
LAPACK: /home/hpages/R/R-3.5.r75051/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] rhdf5_2.26.2

loaded via a namespace (and not attached):
[1] compiler_3.5.1 Rhdf5lib_1.4.2
```

**Comment by grimbough** [comment]
So the expected behaviour is to retain the dimension, but not return anything?

**Comment by hpages** [comment]
Yes. More precisely, it should mimic what `[` does e.g. `h5read("test.h5", "m", list(integer(0), 3:2))` should be the same as `h5read("test.h5", "m")[integer(0), 3:2]` and `h5read("test.h5", "m", list(integer(0), integer(0)))` the same as `h5read("test.h5", "m")[integer(0), integer(0)]`.

Thanks Mike!

**Comment by hpages** [comment]
Also even though the returned array is empty, it should have the expected type. Thanks again.

**Comment by grimbough** [comment]
Hopefully this has been resolved in #36. I can get the version of HDF5Array on Github to pass `R CMD check` now, so hopefully functionality has been restored.  I've also pushed it to Bioc, so should propagate through the build system fairly soon.

**Comment by hpages** [comment]
Tested today with rhdf5 2.27.15. I confirm the issue is gone. Thanks @grimbough
